python/tk_multi_launchapp/base_launcher.py

```diff
@@ -132,22 +255,35 @@ def _launch_app(self, menu_name, app_engine, app_path, app_args, context, versio
             environ_clone = os.environ.copy()
             sys_path_clone = list(sys.path)
 
-            # Get the executable path path and args. Adjust according to
-            # the relevant engine.
-            app_path = apply_version_to_setting(app_path, version)
-            app_args = apply_version_to_setting(app_args, version)
-            if app_engine:
-                (prepped_path, prepped_args) = prepare_launch_for_engine(
-                    app_engine, app_path, app_args, context, file_to_open
+            engine_launcher = self._engine_launcher(app_engine)
+            if sw_version and engine_launcher:
+                launch_args = apply_version_to_setting(app_args, sw_version.version)
+                launch_info = engine_launcher.prepare_launch(
```

Comment:
if this took a simple path instead, could we add it as a sub-clause in `prepare_apps.prepare_launch_for_engine` ([here](https://github.com/shotgunsoftware/tk-multi-launchapp/blob/develop/software_entity/python/tk_multi_launchapp/prepare_apps.py#L17))? Seems this could simplify things a lot!